### PR TITLE
[MRG] Implemented diagonal estimator to solve Issue #7

### PR DESCRIPTION
**Pull request by thomasbazeille**
I added/modified tests for projection scheme and class appropriatly.
New tests pass except once that can fail from time to time but this problem is already patched in PR #4

### CODE REVIEW

File: fmralign/alignment_methods.py [hunk 1]
```diff
@@ -82,6 +83,16 @@ def optimal_permutation(X, Y):
     return permutation
 
 
+def projection(x, y):
+    return np.dot(x, y) / np.linalg.norm(x)**2
```
Comment (bthirion) [inline fmralign/alignment_methods.py]: docstring ?

File: fmralign/alignment_methods.py [hunk 2]
```diff
@@ -82,6 +83,16 @@ def optimal_permutation(X, Y):
...
+
+
+def voxelwise_signal_projection(X, Y, n_jobs):
+    return Parallel(n_jobs)(delayed(projection)(
```
Comment (bthirion) [inline fmralign/alignment_methods.py]: dosctring here too. Also, as it lay not be part of the public API, you may prepend an undrescore:
_voxelwise_signal_projection
Same for the previous function.

File: fmralign/tests/test_alignment_methods.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import numpy as np
 from sklearn.utils.testing import assert_array_almost_equal, assert_greater
 from scipy.linalg import orthogonal_procrustes
-from fmralign.alignment_methods import scaled_procrustes, optimal_permutation, Identity, ScaledOrthogonalAlignment, Hungarian, RidgeAlignment, OptimalTransportAlignment
+from fmralign.alignment_methods import scaled_procrustes, optimal_permutation, voxelwise_signal_projection, Identity, DiagonalAlignment, ScaledOrthogonalAlignment, Hungarian, RidgeAlignment, OptimalTransportAlignment
```
Comment (bthirion) [inline fmralign/tests/test_alignment_methods.py]: pep8: line too long.

File: fmralign/alignment_methods.py (hunk 2 above)
Comment (bthirion) [reply fmralign/alignment_methods.py]: Actually, I'm wondering: you want to directly apply that at the matrix level, avoiding the loop.

Comment (thomasbazeille) [reply fmralign/alignment_methods.py]: After doing the benchmark, this is 4x faster this way:
```
from fmralign.alignment_methods import _voxelwise_signal_projection
import numpy as np
X = np.random.randn(100, 500000)
Y = np.random.randn(100, 500000)
```


`%timeit -n 100 _voxelwise_signal_projection(X, Y, 1)`
> 100 loops, best of 3: 109 ms per loop

`%timeit - n 100 np.sum(X * Y, 1) / np.sum(X * X, 1)`
 > 100 loops, best of 3: 703 ms per loop

**Comment by thomasbazeille** [comment]
Good to merge

**Comment by bthirion** [comment]
Amazing. OK, I owe you a beer ;-) 
B

**Comment by bthirion** [comment]
Thx, merging

**Pull request merged**